src/Resources/contao/library/Contao/System.php

```diff
@@ -314,32 +315,29 @@ public static function loadLanguageFile($strName, $strLanguage=null, $blnNoCache
 		// Load the language(s)
 		foreach ($arrCreateLangs as $strCreateLang)
 		{
-			try
+			// Try to load from cache
+			if (file_exists($kernel->getCacheDir() . '/contao/languages/' . $strCreateLang . '/' . $strName . '.php'))
 			{
-				include $kernel->getContainer()->get('contao.cached_resource_locator')->locate('languages/' . $strCreateLang . '/' . $strName . '.php', null, true);
+				include $kernel->getCacheDir() . '/contao/languages/' . $strCreateLang . '/' . $strName . '.php';
 			}
-			catch (\InvalidArgumentException $e)
+			else
 			{
-				$objLocator = $kernel->getContainer()->get('contao.resource_locator');
-				$xlfLoader  = new XliffFileLoader($kernel->getRootDir(), true);
+				/** @var SplFileInfo[] $files */
+				$files = $kernel->getContainer()->get('contao.resource_finder')->in('languages/' . $strCreateLang)->files()->name($strName . '.php');
```

Comment:
I have not used it here on purpose, because the file locator might throw an exception if there is no file (which is the case for `.php` files). Would have to wrap it in a try catch block. The finder, by contrast, simply returns an empty array.